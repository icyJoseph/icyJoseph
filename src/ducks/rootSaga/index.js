import { all, fork } from "redux-saga/effects";

import { gitHubSaga } from "../github/saga";

export default function* rootSaga() {
  yield all([fork(gitHubSaga)]);
}
