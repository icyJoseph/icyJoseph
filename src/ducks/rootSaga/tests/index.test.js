import rootSaga from "../";

describe("rootSaga", () => {
  const gen = rootSaga().next();
  it("returns a generator with gitHubSaga", () => {
    expect(gen.value).toMatchSnapshot();
  });
});
