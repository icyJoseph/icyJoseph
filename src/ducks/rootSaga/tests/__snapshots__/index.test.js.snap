// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootSaga returns a generator with gitHubSaga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "ALL": Array [
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [],
        "context": null,
        "fn": [Function],
      },
    },
  ],
}
`;
